cache.js

```diff
@@ -4,49 +4,59 @@
  */
 
 import { Image } from 'react-native';
+//undocumented but part of react-native; see
+//https://github.com/facebook/react-native/issues/5603#issuecomment-297959695
+import resolveAssetSource from 'react-native/Libraries/Image/resolveAssetSource';
 
 /**
  * store with
- *  key: imageURL
+ *  key: image
  *  value: {
  *      width: 100,
  *      height: 100,
  *  }
  */
 const cache = new Map();
 
-const getImageSizeFromCache = (imageURL) => {
-    return cache.get(imageURL);
+const getImageSizeFromCache = (image) => {
+    return cache.get(image);
```

Comment:
This `image` is number or `object`. Please use `WeakMap` as `cache`.

Reply:
I don't think we want WeakMap, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap says WeakMap keys need to be objects not primitives. Better to use regular Map as we are now.

I will change the logic though so that we store number and string (the URI) keys rather than number and object because object comparison will fail; see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Using_the_Map_object

Reply:
You are right! 👍